virttest/virt_vm.py

```diff
@@ -1216,6 +1216,9 @@ def print_guest_network_info():
                 if serial:
                     break
                 raise
+            except remote.LoginProcessTerminatedError as err:
```

Comment:
From your paste logs, most of them comes from virt_vm L1373 ,which is wait_for_serial_login(). 
This is not right place to suppress the log

Reply:
When remote.LoginProcessTerminatedError(or ExpectProcessTerminatedError) is raised, it means the aexpect process is dead, why the loop is still running and keeps trying to connect a dead process? 
We should look into why it happens, but not where it happens. I'm not suppressing the log. This patch tries to fix the problem I'm asking above. If anybody gives some reasons why we should not break the loop, then we should fix it in other places by checking the serial condition(This should modify several functions and may break some script, so it has more risks than my current fix).  

In order to show this problem clearly, I'll give more explain.
After LoginProcessTerminatedError happens, the next code runs as below:
self.login->self.get_address(L1089)->self._get_address(L778)->self.address_cache.drop(L829)->raise VMAddressVerificationError(L835)->utils_net.update_mac_ip_address(L1091)->vm.wait_for_serial_login(utils_net.py:3640)->logging.warn("Error occur when update VM )(utils_net.py:3648)->then a new loop again(keep printing above two line logs).

Reply:
LoginProcessTerminatedError inherit from LoginError, which extends generic exception (virttest/remote.py). So LoginProcessTerminatedError throwing DOES NOT mean  'aexpect process is dead' since LoginProcessTerminatedError doesn't implement any additional logic there , but just  override __str__(). 
I think that is why it use generic Exception at last in line 1222.

Reply:
Then we can check the status and output, the make sure it is aexpect.ExpectProcessTerminatedError.  ILoginProcessTerminatedError is raised in handle_prompt when ExpectProcessTerminatedError happens.

Reply:
I checked other places useing LoginProcessTerminatedError, an exceptions.TestError will be raised. I can say LoginProcessTerminatedError has same meaning as aexpect.ExpectProcessTerminatedError. The current code doesn't need to be changed.

Reply:
I find the aexpect code to check the raise condition
```
        # Check if the child has terminated
        if utils_wait.wait_for(lambda: not self.is_alive(), 5, 0, 0.1):
            raise ExpectProcessTerminatedError(patterns, self.get_status(),
                                               output)
```
So that should means only when process is not alive, it will raise this kind of error
So I think it should work @chunfuwen

Reply:
@kylazhang 
@xiaodwan 
aexpect as client ,which essentially call  underlying subprocess.Popen(cmd), and it is hard to know status of cmd accurately.  Suggest combine LoginProcessTerminatedError with previous LoginAuthenticationError , and do some basic test since wait_for_login() is heavily used function.
            except (remote.LoginAuthenticationError,
                    remote.LoginBadClientError, LoginProcessTerminatedError ):
                if serial:
                    break
                raise

Reply:
if serial is true, in wait_for_serial_login, lots of logs will still be printed out. 
I don't think it's a good idea to put together with the two login exceptions.